File: javascript/sdk/claimVmId.js
import { PWRWallet } from "@pwrjs/core";
import dotenv from 'dotenv';
dotenv.config();

// Setting up your wallet in the SDK
const privateKey = process.env.PRIVATE_KEY;
const wallet = new PWRWallet(privateKey);

async function claim() {
    // Add a unique VM ID
    const vmId = 102030;

    // Claim the VM ID
    const txHash = await wallet.claimVmId(vmId);

    // Error handling
    if (txHash.success) {
        console.log("Transaction Hash:", txHash.transactionHash);
    } else {
        console.log("Error:", txHash.message);
    }
}
claim()

File: python/sdk/claim_vmId.py
from pwrpy.pwrwallet import PWRWallet
from dotenv import load_dotenv
import os
load_dotenv()

# Setting up your wallet in the SDK
private_key = os.getenv("PRIVATE_KEY")
wallet = PWRWallet(private_key)

def claim():
    # Add a unique VM ID
    vm_id = 102030

    # Claim the VM ID
    tx_hash = wallet.claim_vm_id(vm_id)

    # Error handling
    if tx_hash.success:
        print("Transaction Hash:", tx_hash.data)
    else:
        print("Error:", tx_hash.message)

if __name__ == "__main__":
    claim()